server/index.js

```diff
@@ -80,6 +80,36 @@ app.post('/api/users/sign-in', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/saved-books/', (req, res, next) => {
+  const { title, author, imageLink, description, buyLink, averageRating, isbn10, category, userId } = req.body;
+  if (!title || !author || !imageLink || !description || !averageRating || !isbn10 || !category || !userId) {
+    throw new ClientError(400, 'missing info');
+  }
+  const insertBookSql = `
+    insert into "books" ("title", "author", "imageLink", "description", "buyLink", "averageRating", "isbn10", "category")
+    values ($1, $2, $3, $4, $5, $6, $7, $8)
+    returning *
+  `;
+  const params = [title, author, imageLink, description, buyLink, averageRating, isbn10, category];
+  db.query(insertBookSql, params)
+    .then(result => {
+      const book = result.rows[0];
+      const bookId = book.bookId;
+      const insertUserBookSql = `
+    insert into "usersAddedBooks" ("userId", "bookId")
+    values ($1, $2)
+    returning *
+  `;
+      const insertUserBookParams = [userId, bookId];
+      db.query(insertUserBookSql, insertUserBookParams)
+        .then(result => {
```

Comment:
be sure to `return` this promise chain so that the final `.catch()` will get errors.

```js
return db.query(...)
```

Reply:
<img width="552" alt="Screen Shot 2022-09-19 at 1 40 19 PM" src="https://user-images.githubusercontent.com/106217355/191112359-fce72801-4565-4ce7-b001-5f7c06fa4f49.png">